doc/source/whatsnew/v1.2.0.rst

```diff
@@ -293,6 +293,7 @@ Indexing
 
 - Bug in :meth:`PeriodIndex.get_loc` incorrectly raising ``ValueError`` on non-datelike strings instead of ``KeyError``, causing similar errors in :meth:`Series.__geitem__`, :meth:`Series.__contains__`, and :meth:`Series.loc.__getitem__` (:issue:`34240`)
 - Bug in :meth:`Index.sort_values` where, when empty values were passed, the method would break by trying to compare missing values instead of pushing them to the end of the sort order. (:issue:`35584`)
+- Harmonize resulting index names from :meth:`Index.union` and :meth:`Index.intersection` across various index types (:issue:`35847`)
```

Comment:
if there are significant changes that a user would want to know can you add that detail

Reply:
also migth warrant a sub-section if this is something easier to see via code-examples

Reply:
yup, added in a bit about name preservation

Reply:
right add the issue number above and remove this note